azure-servicemanagement-legacy/tests/test_legacy_mgmt_misc.py

```diff
@@ -2143,32 +2143,29 @@ def test_list_os_images_public(self):
         self.assertIsNotNone(result)
         self.assertTrue(len(result) > 0)
 
-        image = None
+        image_attributes = {}
         for temp in result:
             self.assertIn(temp.category, ['User', 'Public', 'Private', 'MSDN'])
-            if temp.category == 'Public':
-                image = temp
-                break
-
-        self.assertIsNotNone(image)
-        self.assertGreater(len(image.category), 0)
-        self.assertGreater(len(image.label), 0)
-        self.assertGreater(len(image.location), 0)
-        self.assertIsNotNone(image.logical_size_in_gb)
-        self.assertGreaterEqual(image.logical_size_in_gb, 0)
-        self.assertGreater(len(image.name), 0)
-        self.assertGreater(len(image.os), 0)
-        self.assertIsNotNone(image.eula)
-        self.assertGreater(len(image.description), 0)
-        self.assertGreater(len(image.image_family), 0)
-        self.assertIsNotNone(image.show_in_gui)
-        self.assertGreater(len(image.published_date), 0)
-        self.assertIsNotNone(image.is_premium)
-        self.assertIsNotNone(image.icon_uri)
-        self.assertIsNotNone(image.privacy_uri)
-        self.assertIsNotNone(image.recommended_vm_size)
-        self.assertGreater(len(image.publisher_name), 0)
-        self.assertIsNotNone(image.small_icon_uri)
+            for attr in temp.__dict__:
+                if getattr(temp, attr):
+                    image_attributes[attr] = getattr(temp, attr)
+
+        self.assertTrue(image_attributes['category'])
```

Comment:
@colemickens I updated the test to extract every non-False (False in Python being empty string, 0, etc.) attributes from all `OSImage` instances in the list result. Considering all images in Azure, there is always at least one image with a given attribute (even if there are almost all optional). This allows us to test the parser, because by default all field are False (empty string or 0). If the parser misses to parse something (like it did with `recommended_vm_size`), the test will fail.